File: Java/Hard/WrongAnswer/total-appeal-of-a-string_5.java
class Solution {
    public long appealSum(String s) {
        
        long ans = 0;
        int n = s.length();
        int[] count = new int[26];
        for (int i = 0; i < n; i++) {
            int[] temp = new int[26];
            for (int j = i; j < n; j++) {
                temp[s.charAt(j) - 'a']++;
                if (temp[s.charAt(j) - 'a'] == 1) {
                    count[s.charAt(j) - 'a']++;
                }
                ans += count[s.charAt(j) - 'a'];
            }
        }
        return ans;
    }
}

File: C++/Hard/TimeLimitExceeded/total-appeal-of-a-string_5.cpp
class Solution {
public:
    long long appealSum(string s) {
        
        long long result = 0;
        int n = s.length();
        for (int i = 0; i < n; i++) {
            int count[26] = {0};
            int k = 0;
            for (int j = i; j < n; j++) {
                int index = s[j] - 'a';
                if (count[index] == 0) {
                    count[index] = 1;
                    k++;
                }
                result += k;
            }
        }
        return result;
    }
};